quasar-core/src/main/java/co/paralleluniverse/strands/StrandFactoryBuilder.java

```diff
@@ -195,10 +198,9 @@ public Strand newStrand(SuspendableCallable<?> target) {
 
     static {
         try {
-            nextThreadNum = Thread.class.getDeclaredMethod("nextThreadNum");
-            nextThreadNum.setAccessible(true);
-        } catch (ReflectiveOperationException e) {
-            throw new AssertionError(e);
+            nextThreadNum = doPrivileged(new GetAccessDeclaredMethod(Thread.class, "nextThreadNum"));
+        } catch (PrivilegedActionException e) {
```

Comment:
Have you double checked this couldn't change the behaviour if any other exception gets thrown?

Reply:
[According to the JavaDoc](https://docs.oracle.com/javase/8/docs/api/java/security/AccessController.html#doPrivileged-java.security.PrivilegedExceptionAction-), `PrivilegedActionException` is only thrown to wrap a _checked_ exception. Therefore Java's compilation rules _require_ that the underlying behaviour has not changed.